src/pages/Dashboard/MyAccount/MyAccountUi.tsx

```diff
@@ -0,0 +1,217 @@
+          <Tooltip css={styles.tooltip} title={t('myAccount.apyWithXvsTooltip')}>
+            <Icon css={styles.infoIcon} name="info" />
+          </Tooltip>
+
+          <Typography color="text.primary" variant="small1" css={styles.apyWithXvsLabel}>
+            {t('myAccount.apyWithXvs')}
+          </Typography>
+
+          <Toggle css={styles.toggle} value={withXvs} onChange={handleXvsToggleChange} />
+        </Typography>
+      </div>
+
+      <div css={styles.netApyContainer}>
+        <div css={styles.netApy}>
+          <Typography component="div" variant="small2" css={styles.netApyLabel}>
+            {t('myAccount.netApy')}
+          </Typography>
+
+          <Tooltip css={styles.tooltip} title={t('myAccount.netApyTooltip')}>
+            <Icon css={styles.infoIcon} name="info" />
+          </Tooltip>
+        </div>
+
+        <Typography variant="h1" color="interactive.success">
+          {typeof netApyPercentage === 'number' ? `${netApyPercentage}%` : '-'}
+        </Typography>
+      </div>
+
+      <ul css={styles.list}>
+        <Typography component="li" variant="h4" css={styles.item}>
+          <Typography component="div" variant="small2" css={styles.labelListItem}>
+            {t('myAccount.dailyEarnings')}
+          </Typography>
+
+          {typeof dailyEarningsCents === 'number'
+            ? formatCentsToReadableValue(dailyEarningsCents)
+            : '-'}
+        </Typography>
+
+        <Typography component="li" variant="h4" css={styles.item}>
+          <Typography component="div" variant="small2" css={styles.labelListItem}>
+            {t('myAccount.supplyBalance')}
+          </Typography>
+
+          {typeof supplyBalanceCents === 'number'
+            ? formatCentsToReadableValue(supplyBalanceCents)
+            : '-'}
+        </Typography>
+
+        <Typography component="li" variant="h4" css={styles.item}>
+          <Typography component="div" variant="small2" css={styles.labelListItem}>
+            {t('myAccount.borrowBalance')}
+          </Typography>
+
+          {readableBorrowBalance || '-'}
+        </Typography>
+      </ul>
+
+      <div css={[styles.row, styles.topProgressBarLegend]}>
+        <div css={styles.inlineContainer}>
+          <Typography component="span" variant="small2" css={styles.inlineLabel}>
+            {t('myAccount.borrowLimitUsed')}
+          </Typography>
+
+          <Typography component="span" variant="small1" color="text.primary">
+            {readableBorrowLimitUsedPercentage || '-'}
+          </Typography>
+        </div>
+
+        <div css={styles.inlineContainer}>
+          <Typography component="span" variant="small2" css={styles.inlineLabel}>
+            {t('myAccount.limit')}
+          </Typography>
+
+          <Typography component="span" variant="small1" color="text.primary">
+            {typeof borrowLimitCents === 'number'
+              ? formatCentsToReadableValue(borrowLimitCents, true)
```

Comment:
This is the same duplicated logic where we check if a value is a number, if it is we do something, if it isn't we return a `-`